src/dbsync/src/sqlite/sqlite_dbengine.cpp

```diff
@@ -820,41 +830,54 @@ bool SQLiteDBEngine::deleteRows(const std::string& table,
     return ret;
 }
 
-void SQLiteDBEngine::deleteRows(const std::string& table,
-                                const nlohmann::json& data,
-                                const std::vector<std::string>& primaryKeyList)
+void SQLiteDBEngine::deleteRowsbyPK(const std::string& table,
+                                    const nlohmann::json& data)
 {
-    const auto& tableFields { m_tableFields[table] };
-    const auto& stmt
+    std::vector<std::string> primaryKeyList;
+    if (getPrimaryKeysFromTable(table, primaryKeyList))
     {
-        getStatement(buildDeleteBulkDataSqlQuery(table, primaryKeyList))
-    };
+        const auto& tableFields { m_tableFields[table] };
+        const auto& stmt
+        {
+            getStatement(buildDeleteBulkDataSqlQuery(table, primaryKeyList))
+        };
 
-    for (const auto& jsRow : data)
-    {
-        int32_t index { 1l };
-        for (const auto& pkValue : primaryKeyList)
+        for (const auto& jsRow : data)
         {
-            const auto& it
+            int32_t index { 1l };
+            for (const auto& pkValue : primaryKeyList)
             {
-                std::find_if(tableFields.begin(), tableFields.end(),
-                                [&pkValue](const ColumnData& column)
-                                {
-                                    return 0 == std::get<Name>(column).compare(pkValue);
-                                })
-            };
+                const auto& it
+                {
+                    std::find_if(tableFields.begin(), tableFields.end(),
+                                    [&pkValue](const ColumnData& column)
+                                    {
+                                        return 0 == std::get<Name>(column).compare(pkValue);
+                                    })
+                };
 
-            if(it != tableFields.end())
-            {
-                bindJsonData(stmt, *it, jsRow, index);
-                ++index;
+                if(it != tableFields.end())
+                {
+                    bindJsonData(stmt, *it, jsRow, index);
+                    ++index;
+                }
             }
+            stmt->step();
+            stmt->reset();
         }
-        stmt->step();
-        stmt->reset();
     }
 }
 
+void SQLiteDBEngine::deleteRowsByFilter(const std::string& table,
+                                        const nlohmann::json::const_iterator& itFilter)
+{
+    const auto& stmt
+    {
+        getStatement("DELETE FROM "+table+" "+itFilter->get<std::string>())
```

Comment:
We should not use the getStatement in queries that have data/filter values, since getStatement() caches the statements, this would cause many unnecessary allocations increasing the footprint.

Reply:
you can use execute() for the use case with filter pre-defined in the input json

Reply:
We could prevent from this implementation the possibility of doing a "delete from table  ;" the purpose of this interface is to be able to eliminate as atomically as possible, and if the filter has a space it can delete the content of the entire table. 
The issue is lacking in information, it is not your fault.

Basically the change is to pre-define the WHERE clause in the filter query.

Reply:
Thanks for your comments @Dwordcito.
- Before calling the `deleteRowByFilter()` method I'm checking that the itFilter exists and, moreover, that it is not empty, so the "delete from table ;" case will not happen.
- Agree with the execute instead of the getStatement, changing this right away.